src/three/math/Ellipsoid.js

```diff
@@ -200,4 +203,43 @@ export class Ellipsoid {
 
 	}
 
+	calculateHorizonDistance( latitude, elevation ) {
+
+		// elevation relative to the surface of the ellipsoid at the specified latitude
+		const effectiveRadius = this.calculateEffectiveRadius( latitude );
+		return Math.sqrt( 2 * effectiveRadius * elevation + elevation ** 2 );
+
+	}
+
+	calculateEffectiveRadius( latitude ) {
+
+		const semiMajorAxis = this.radius.x;
+		const semiMinorAxis = this.radius.z;
+		const eSquared = 1 - ( semiMinorAxis ** 2 / semiMajorAxis ** 2 );
+		const phi = latitude * MathUtils.DEG2RAD;
+
+		const sinPhiSquared = Math.sin( phi ) ** 2;
+		const N = semiMajorAxis / Math.sqrt( 1 - eSquared * sinPhiSquared );
+		return N;
+
+	}
+
+	worldToLocal( pos ) {
+
+		pos.applyQuaternion( toLocalQuaternion );
+
+	}
+
+	getPositionElevation( pos ) {
+
+		_pos.copy( pos );
+		this.worldToLocal( _pos );
+
+		this.getPositionToSurfacePoint( _pos, _vec3 );
+		const elevation = _vec3.distanceTo( _pos );
+
+		return elevation;
```

Comment:
There shouldn't be any notion of a transform here - the 3d point should be passed in the same frame that the ellipsoid is stored in. If we assume a 90 degree rotation here then if the rotation of the globe object is set to anything different then this will break.

The 90 globe rotation is set on the tile root group [here](https://github.com/NASA-AMMOS/3DTilesRendererJS/blob/5d156142485764e1c1d64d3d024f8e7a9efb6edb/example/googleMapsExample.js#L52). So calling this function should look like this:

```js
_invMatrix.copy( tilesRenderer.group.matrixWorld ).invert();
_tempVec.copy( camera.position ).applyMatrix4( _invMatrix );
const elevation = ellipsoid.getPositionElevation();
```

The same pattern is used in GlobeControls [here](https://github.com/NASA-AMMOS/3DTilesRendererJS/blob/5d156142485764e1c1d64d3d024f8e7a9efb6edb/src/three/controls/GlobeControls.js#L101-L103).